cpp/basix/finite-element.cpp

```diff
@@ -335,6 +335,78 @@ FiniteElement::FiniteElement(
     throw std::runtime_error(
         "Number of entity dofs does not match total number of dofs");
   }
+
+  // Check if base transformations are all permutations
+  _dof_transformations_are_permutations = true;
+  _dof_transformations_are_identity = true;
+  for (std::size_t i = 0; _dof_transformations_are_permutations
+                          && i < _entity_transformations.size();
+       ++i)
+  {
+    for (std::size_t row = 0; row < _entity_transformations[i].shape(0); ++row)
+    {
+      double rmin
+          = xt::amin(xt::view(_entity_transformations[i], row, xt::all()))(0);
+      double rmax
+          = xt::amax(xt::view(_entity_transformations[i], row, xt::all()))(0);
+      double rtot
+          = xt::sum(xt::view(_entity_transformations[i], row, xt::all()))(0);
+      if ((_entity_transformations[i].shape(1) != 1
```

Comment:
These values look a bit arbitrary. Is these a neater way?

Reply:
I've made them use an `isClose` function to make it more readable but can't find a better way to compare two floats. For all the elements so far the nearest we get to 1 or 0 without being equal is 0.5, so the large epsilon comparison is good enough for now

Reply:
`xt::close` [1] not suitable?

[1] https://xtensor.readthedocs.io/en/latest/api/classif_functions.html#_CPPv4I00EN2xt7iscloseEDaRR2E1RR2E2ddb